### PR TITLE
add in annual work history charts to insights page

### PR DESCRIPTION
**Pull request by tom0827**
#2280 
Add front end and back end for annual work creation/closure charts

### CODE REVIEW

File: epictrack-api/src/api/actions/set_work_state.py [hunk 1]
```diff
@@ -25,8 +26,10 @@ def run(self, source_event, params) -> None:
             }
             change_phase_end_event.run(source_event, change_phase_end_event_param)
         is_active = True
+        ended_at = None
         if work_state in [state.value for state in EndingWorkStateEnum]:
             is_active = False
+            ended_at = datetime.now()
         db.session.query(Work).filter(Work.id == source_event.work_id).update(
-            {Work.work_state: work_state, Work.is_active: is_active}
+            {Work.work_state: work_state, Work.is_active: is_active, Work.ended_at: ended_at}
```
Comment (dinesh-aot) [inline epictrack-api/src/api/actions/set_work_state.py]: can't you use "work_decision_date" column for this purpose? I can see that this column is not used anywhere. But the plan is to set it when the decision is made on that work.

Comment (tom0827) [reply epictrack-api/src/api/actions/set_work_state.py]: yea i see its not used anywhere. what decision is it reffering to?

Comment (dinesh-aot) [reply epictrack-api/src/api/actions/set_work_state.py]: Whenever the work state changes to one of the ending_states

Comment (tom0827) [reply epictrack-api/src/api/actions/set_work_state.py]: oh ok. ill refactor to use that

**Review by dinesh-aot — APPROVED** [review]

**Pull request merged**